src/jit/codegencommon.cpp

```diff
@@ -3165,11 +3165,17 @@ void CodeGen::genGenerateCode(void** codePtr, ULONG* nativeSizeOfCode)
     /* Check our max stack level. Needed for fgAddCodeRef().
        We need to relax the assert as our estimation won't include code-gen
        stack changes (which we know don't affect fgAddCodeRef()) */
-    noway_assert(getEmitter()->emitMaxStackDepth <=
-                 (compiler->fgPtrArgCntMax +              // Max number of pointer-sized stack arguments.
-                  compiler->compHndBBtabCount +           // Return address for locally-called finallys
-                  genTypeStSz(TYP_LONG) +                 // longs/doubles may be transferred via stack, etc
-                  (compiler->compTailCallUsed ? 4 : 0))); // CORINFO_HELP_TAILCALL args
+    {
+        unsigned accStackDepth = compiler->fgPtrArgCntMax +    // Max number of pointer-sized stack arguments.
+                                 compiler->compHndBBtabCount + // Return address for locally-called finallys
+                                 genTypeStSz(TYP_LONG) +       // longs/doubles may be transferred via stack, etc
+#if defined(UNIX_X86_ABI)
+                                 (genTypeStSz(TYP_INT) * 3) + // stack align for x86 0 to 3 padding
+#endif
```

Comment:
I would prefer that we just write this as two statements, with  the second statement being:

```
#if defined(UNIX_X86_ABI)
                                accStackDepth  += (genTypeStSz(TYP_INT) * 3);     // stack align for x86 allow up to 3 INT for padding
#endif
```